test/struct-skip.js

```diff
@@ -0,0 +1,170 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var test = require('tape');
+
+var Buffer = require('buffer').Buffer;
+var Spec = require('../spec');
+var fs = require('fs');
+var path = require('path');
+var source = fs.readFileSync(path.join(__dirname, 'struct.thrift'), 'ascii');
+var spec = new Spec({source: source});
+
+var Health = spec.$Health;
+
+test('skip void', function t(assert) {
+    var result = Health.rw.readFrom(new Buffer([
+        0x02,                     // type:1   -- 2 -- BOOL
+        0x00, 0x01,               // id:2     -- 1 -- WHAT EVEN IS!?
+        0x01,                     // typeid:1 -- 1 -- VOID
+        0x00                      // typeid:1 -- 0 -- STOP
+    ]), 0);
+    if (result.err) {
+        return assert.end(result.err);
+    }
+    assert.deepEqual(result.value, new Health());
+    assert.end();
+});
+
+test('string', function t(assert) {
+    var result = Health.rw.readFrom(new Buffer([
+        0x02,                     // type:1   -- 2  -- BOOL
+        0x00, 0x01,               // id:2     -- 1  -- WHAT EVEN IS!?
+        11,                       // typeid:1 -- 11 -- STRING
+        0x00, 0x00, 0x00, 0x02,   // len~4
+        0x20, 0x20,               // '  '
+        0x00                      // typeid:1 -- 0  -- STOP
+    ]), 0);
+    if (result.err) {
+        return assert.end(result.err);
+    }
+    assert.deepEqual(result.value, new Health());
+    assert.end();
+});
+
+test('struct', function t(assert) {
+    var result = Health.rw.readFrom(new Buffer([
+        0x02,                     // type:1   -- 2  -- BOOL
+        0x00, 0x01,               // id:2     -- 1  -- WHAT EVEN IS!?
+        12,                       // typeid:1 -- 12 -- STRUCT
+        0x01,                     // typeid:1 -- 1  -- VOID
+        0x01,                     // typeid:1 -- 1  -- VOID
+        11,                       // typeid:1 -- 11 -- STRING
+        0x00, 0x00, 0x00, 0x02,   // len~4
+        0x20, 0x20,               // '  '
+        0x01,                     // typeid:1 -- 1  -- VOID
+        0x00,                     // typeid:1 -- 0  -- STOP
+        0x00                      // typeid:1 -- 0  -- STOP
+    ]), 0);
+    if (result.err) {
+        return assert.end(result.err);
+    }
+    assert.deepEqual(result.value, new Health());
+    assert.end();
+});
+
+test('map', function t(assert) {
+    var result = Health.rw.readFrom(new Buffer([
+        0x02,                     // type:1           -- 2 BOOL
+        0x00, 0x01,               // id:2             -- 1 UNKNOWN
+        0x0d,                   // typeid:1           -- 13, map
+
+        // Thus begins a large map
```

Comment:
I borrowed these byte sequences from map and list tests. Consider importing them by reference from a library of bytes.